src/components/select/filterable-select/filterable-select.stories.mdx

```diff
@@ -0,0 +1,134 @@
+It is still possible to pass `onChange` to hook a callback for state changes, however to set the initial value, `defaultValue` can be passed.
+
+```jsx
+import { FilterableSelect, Option } from "carbon-react/lib/components/select";
+
+const MyComponent = () => (
+  <FilterableSelect onChange={ onChange } defaultValue='2' id='selectId' name='selectName'>
+    <Option text='Amber' value='1' />
+    <Option text='Black' value='2' />
+    <Option text='Blue' value='3' />
+  </FilterableSelect>
+);
+```
+
+### Basic usage:
+
+<Preview>
+  <Story name="basic" parameters={{ info: { disable: true }}} >
+    <div style={{ height: '250px' }}>
+      <FilterableSelect name='simple' id='simple' label='label' labelInline>
+        <Option text='Amber' value='1' />
+        <Option text='Black' value='2' />
+        <Option text='Blue' value='3' />
+        <Option text='Brown' value='4' />
+        <Option text='Green' value='5' />
+        <Option text='Orange' value='6' />
+        <Option text='Pink' value='7' />
+        <Option text='Purple' value='8' />
+        <Option text='Red' value='9' />
+        <Option text='White' value='10' />
+        <Option text='Yellow' value='11' />
+      </FilterableSelect>
+    </div>
+  </Story>
+</Preview>
+
+### Disabled:
+
+<Preview>
+  <Story name="disabled" parameters={{ info: { disable: true }}} >
+    <div style={{ height: '250px' }}>
+      <FilterableSelect name='disabled' id='disabled' defaultValue='3' disabled>
+        <Option text='Amber' value='1' />
+        <Option text='Black' value='2' />
+        <Option text='Blue' value='3' />
+        <Option text='Brown' value='4' />
+        <Option text='Green' value='5' />
+        <Option text='Orange' value='6' />
+        <Option text='Pink' value='7' />
+        <Option text='Purple' value='8' />
+        <Option text='Red' value='9' />
+        <Option text='White' value='10' />
+        <Option text='Yellow' value='11' />
+      </FilterableSelect>
+    </div>
+  </Story>
+</Preview>
+
+### Read Only:
+
+<Preview>
+  <Story name="readonly" parameters={{ info: { disable: true }}} >
+    <div style={{ height: '250px' }}>
+      <FilterableSelect name='readonly' id='readonly' defaultValue='4' readOnly>
+        <Option text='Amber' value='1' />
+        <Option text='Black' value='2' />
+        <Option text='Blue' value='3' />
+        <Option text='Brown' value='4' />
+        <Option text='Green' value='5' />
+        <Option text='Orange' value='6' />
+        <Option text='Pink' value='7' />
+        <Option text='Purple' value='8' />
+        <Option text='Red' value='9' />
+        <Option text='White' value='10' />
+        <Option text='Yellow' value='11' />
+      </FilterableSelect>
+    </div>
+  </Story>
+</Preview>
+
+## Props:
+
+<Props of={ FilterableSelect } />
+
+### Props derived from the Textbox Component
+
+<Props of={ SelectTextbox } exclude={ ['onKeyDown', 'formattedValue', 'selectedValue'] }/>
```

Comment:
Is there a way to actually remove these props from the component? Or is this the only way because we still have the `__experimental__` select component? I'm wary that we could add more props to Textbox and have them surfaced here unintentionally.

I think it would be best if we specifically defined the prop types for the select component and passed the props by key/value instead of spreading the props.

Reply:
all these props are not taken directly from the Textbox, so you should not be worried about new ones surfacing. If we specify them then we'll have to add 40 lines of code per every select based component (SimpleSelect, FilterableSelect, MultiSelect).